azure-iot-device/tests/provisioning/test_sync_provisioning_device_client.py

```diff
@@ -176,30 +97,34 @@ def mock_polling_machine_success(self, mocker):
 
     @pytest.mark.it("Register calls register on polling machine with passed in callback")
     def test_client_register_success_calls_polling_machine_register_with_callback(
-        self, mocker, mock_polling_machine_success, params_clients
+        self, mocker, mock_polling_machine_success
     ):
-        state_based_mqtt = mocker.MagicMock()
-        mock_polling_machine_init = mocker.patch(params_clients["patch_string"])
-        # TODO After deciding on one client, no need for params and patch will become
-        # mock_polling_machine_init = mocker.patch(
-        #     "azure.iot.device.provisioning.provisioning_device_client.PollingMachine"
-        # )
+        mqtt_provisioning_pipeline = mocker.MagicMock()
+        mock_polling_machine_init = mocker.patch(
+            "azure.iot.device.provisioning.provisioning_device_client.PollingMachine"
+        )
         mock_polling_machine_init.return_value = mock_polling_machine_success
 
-        client = params_clients["client_class"](state_based_mqtt)
+        client = ProvisioningDeviceClient(mqtt_provisioning_pipeline)
         client.register()
         assert mock_polling_machine_success.register.call_count == 1
+        assert isinstance(
+            mock_polling_machine_success.register.call_args[1]["callback"], types.FunctionType
+        )
```

Comment:
I found a new builtin that is a little better.  You can say `assert callable( mock_polling_machine_success.register.call_args[1]["callback"])` because it supports things that are callable but not necessarily functions.  

(I'm only mentioning it because I came across it on a thread yesterday and I was excited to find it)

Reply:
i changed in these tests as of now. Yes it is a better approach. 

---
In reply to: [294444198](https://github.com/Azure/azure-iot-sdk-python-preview/pull/109#discussion_r294444198) [](ancestors = 294444198)